ponyc: 0.42.0 -> 0.44.0, pony-corral: 0.5.3 -> 0.5.4, pony-stable: removal

redvers opened the pull request:
###### Motivation for this change

Updating the ponylang compiler to the latest version:

* Fix MacOS version mismatch warnings when linking pony programs.
* Fix calculation of "is prime" for numbers > 1321.
* Prevent non-opaque structs from being used as behaviour parameters.
* Update to LLVM 12.0.1.
* Added FileMode.u32.
* Changes to Promises Library.
* Remove zlib from libponyc as a dependency.
* Fix compile-time crash related to Pony-specific Optimizations.
* Update FilePath constructors to allow a non-partial way to create a FilePath.

###### Things done

* Updated ponyc to 0.44.0
* Updated googlebenchmark dependency to 1.5.4
* Updated googletest to 1.10.0
* Replaced a fetchurl call with a fetchFromGitHub call.

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->


<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

redvers committed:
ponyc: 0.42.0 -> 0.44.0

* Fix MacOS version mismatch warnings when linking pony programs.
* Fix calculation of "is prime" for numbers > 1321.
* Prevent non-opaque structs from being used as behaviour parameters.
* Update to LLVM 12.0.1.
* Added FileMode.u32.
* Changes to Promises Library.
* Remove zlib from libponyc as a dependency.
* Fix compile-time crash related to Pony-specific Optimizations.
* Update FilePath constructors to allow a non-partial way to create a FilePath.

Updated googlebenchmark dependency to 1.5.4
Updated googletest to 1.10.0

Replaced a fetchurl call with a fetchFromGitHub call.

SuperSandro2000 commented on pkgs/development/compilers/ponyc/default.nix:
```diff
@@ -26,22 +26,23 @@ stdenv.mkDerivation (rec {
-        sha256 = "17147961i01fl099ygxjx4asvjanwdd446nwbq9v8156h98zxwcv";
+      googletest = fetchFromGitHub {
+        owner = "google";
+        repo = "googletest";
```
Can we use the googletest from nixpkgs?

redvers replied:
Great question.  I've seen upstream bump and unbump the versions for this due to issues and I'm not sure why.

I'd be hesitant to link to use the nixpkgs one unless I could pin it to a version.

Let me reach out to the developers to see if I can understand why there were issues with just tracking google and I'll get back to you.

Thank you! <3

SuperSandro2000 replied:
> I'd be hesitant to link to use the nixpkgs one unless I could pin it to a version.

You can overwrite the version if you want.

redvers replied:
The reason for the set version is that gtest versions higher than the one they specified are broken on freebsd (which is a supported platform for the language).

It is a build-time dependency only so as long as it builds on the platforms that nix supports there's no reason why we can't use nix's version.

(Obviously - if there was a nix on freebsd that would be a different question)

I can make that change now.

redvers replied:
(My medium-term goal is to try and get this package to the point where the r-ryantm bot will be able to do updates independent of me)

SuperSandro2000 replied:
bsd support is not really fleshed out for nix yet. I would worry about that when it is here.

veprbl commented:
Result of `nixpkgs-review pr 140189` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>pony-corral</li>
    <li>pony-stable</li>
  </ul>
</details>
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>ponyc</li>
  </ul>
</details>

veprbl commented:
@GrahamcOfBorg build pony-corral pony-stable

redvers commented:
> Result of `nixpkgs-review pr 140189` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
> 2 packages failed to build:
> 
>     * pony-corral
>     * pony-stable

pony-stable has been deprecated for a while (pony-corral is the replacement).
pony-corral absolutely should build.

Question - they're different entries in top-level but live in the same nixpkgs directory and should always be in sync.  Is it appropriate to remove pony-stable and update pony-corral in the same PR?

I always thought they were meant to be separate processes?

SuperSandro2000 commented:
> Is it appropriate to remove pony-stable and update pony-corral in the same PR?

Yeah, sure.

redvers commented:
> > Is it appropriate to remove pony-stable and update pony-corral in the same PR?
> 
> Yeah, sure.

To make sure I understand correctly - adding it to this PR is okay?

SuperSandro2000 commented:
> To make sure I understand correctly - adding it to this PR is okay?

Not a problem for me. It just can take a little it longer until we get this PR merged.

redvers commented:
This just triggered a new release of corral.

I guess the subject of the PR would need to change to:
ponyc: 0.42.0->0.44.0, corral: 0.5.3->0.5.4, pony-stable: removed

Does that make sense?

redvers committed:
pony-stable: removal

pony-stable was deprecated upstream on May 12 2020
and we replaced by pony-corral.

redvers committed:
pony-corral: 0.5.3 -> 0.5.4

* Update to compile with Pony 0.44.0

redvers commented:
Result of `nixpkgs-review pr 140189` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>pony-corral</li>
    <li>ponyc</li>
  </ul>
</details>

veprbl commented:
Result of `nixpkgs-review pr 140189` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>pony-corral</li>
    <li>ponyc</li>
  </ul>
</details>
